File: examples-sdk/typescript/304-rag-citation-extraction.ts
#!/usr/bin/env tsx
// 304-rag-citation-extraction — TypeScript SDK equivalent
// Compile: vil compile --from typescript --input 304-rag-citation-extraction.ts --release

import { VilPipeline, VilServer, ServiceProcess } from "vil-sdk";

const server = new VilServer("rag-citation-extraction", 3113);
const rag_citation = new ServiceProcess("rag-citation");
rag_citation.endpoint("POST", "/cited-rag", "cited_rag_handler");
server.service(rag_citation);
server.compile();

File: examples-sdk/python/304-rag-citation-extraction.py
#!/usr/bin/env python3
"""304-rag-citation-extraction — Python SDK equivalent
Compile: vil compile --from python --input 304-rag-citation-extraction.py --release
"""
import os
from vil import VilPipeline, VilServer, ServiceProcess

server = VilServer("rag-citation-extraction", port=3113)
rag_citation = server.service_process("rag-citation")
rag_citation.endpoint("POST", "/cited-rag", "cited_rag_handler")
server.compile()
